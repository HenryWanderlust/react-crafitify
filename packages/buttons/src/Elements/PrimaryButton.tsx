import React from 'react';
import styled from 'styled-components';

interface PrimaryButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  onClick?: () => void;
  disabled?: boolean;
}

const PrimaryButtonWrapper = styled.button<PrimaryButtonProps>`
  padding: 8px 16px;
  font-size: 16px;
  background-color: var(--primary-color, #007bff);
  color: var(--button-text-color, white);
  border: none;
  border-radius: 4px;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? 0.5 : 1)};
`;

const PrimaryButton: React.FC<PrimaryButtonProps> = ({ onClick, disabled, children, ...rest }) => {
  return (
    <PrimaryButtonWrapper onClick={onClick} disabled={disabled} {...rest}>
      {children}
    </PrimaryButtonWrapper>
  );
};

export { PrimaryButton, PrimaryButtonProps };
