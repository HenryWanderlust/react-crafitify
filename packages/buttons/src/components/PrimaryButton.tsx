// src/components/PrimaryButton.tsx

import React from 'react';
import styled from 'styled-components';

interface PrimaryButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  onClick?: () => void;
  disabled?: boolean;
  className?: string;
}

const PrimaryButtonWrapper = styled.button<PrimaryButtonProps>`
  padding: 8px 16px;
  font-size: 16px;
  background-color: #007bff;
  color: white;
  border: none;
  border-radius: 4px;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? 0.5 : 1)};
`;

const PrimaryButton: React.FC<PrimaryButtonProps> = ({ onClick, disabled, className, children }) => {
  return (
    <PrimaryButtonWrapper onClick={onClick} disabled={disabled} className={className}>
      {children}
    </PrimaryButtonWrapper>
  );
};

export { PrimaryButton, PrimaryButtonProps }; // Export the PrimaryButton and PrimaryButtonProps
