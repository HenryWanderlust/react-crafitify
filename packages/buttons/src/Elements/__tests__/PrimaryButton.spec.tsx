import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // Import this to use jest-dom matchers

import { PrimaryButton } from '../PrimaryButton';

describe('PrimaryButton', () => {
  test('renders the button with the default label', () => {
    const { getByText } = render(<PrimaryButton>Click Me</PrimaryButton>);
    expect(getByText('Click Me')).toBeInTheDocument();
  });

  test('renders the button as disabled', () => {
    const { getByText } = render(<PrimaryButton disabled>Disabled Button</PrimaryButton>);
    expect(getByText('Disabled Button')).toBeDisabled();
  });

  test('renders the button with custom class name', () => {
    const { getByText, container } = render(
      <PrimaryButton className="custom-button">Custom Style</PrimaryButton>
    );
    const button = getByText('Custom Style');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('custom-button');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls the custom onClick handler when clicked', () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <PrimaryButton onClick={onClickMock}>Custom OnClick</PrimaryButton>
    );
    const button = getByText('Custom OnClick');
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
