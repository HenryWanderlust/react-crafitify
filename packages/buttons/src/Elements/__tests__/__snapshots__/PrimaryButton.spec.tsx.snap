// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders the button with custom class name 1`] = `
<button
  class="sc-bcPKhP fBJcTv custom-button"
>
  Custom Style
</button>
`;
